pkg/kubectl/describe.go

```diff
@@ -508,7 +507,7 @@ func describePod(pod *api.Pod, events *api.EventList) (string, error) {
 			}
 		}
 		describeVolumes(pod.Spec.Volumes, out, "")
-		fmt.Fprintf(out, "QoS Class:\t%s\n", qos.InternalGetPodQOS(pod))
+		fmt.Fprintf(out, "QoS Class:\t%s\n", pod.Status.QOSClass)
```

Comment:
To clarify my question, I wonder if this should be a field on the PodSpec that is system managed and not on the PodStatus since it's immutable.  In that regard, I can see the effective qos class without scheduling.